src/ast/passes/semantic_analyser.cpp

```diff
@@ -1178,6 +1224,33 @@ void SemanticAnalyser::visit(Call &call)
 
     call.type = CreateUInt64();
   }
+  else if (call.func == "offsetof")
+  {
+    if (check_nargs(call, 2))
+    {
+      // Handle Identifier only, see visit(Call) of offsetof()
+      auto element = dynamic_cast<Identifier *>(call.vargs->at(1));
+      if (element)
```

Comment:
Similar to above, probably better to use an `assert()` here to document the invariant. It's an internal bug if the first arg is not an `Identifier`